internal/gateway/backend/lease_actions.go

```diff
@@ -153,30 +163,38 @@ func (s *Services) CommitLease(ctx context.Context, tokenStr, oldRootHash, newRo
 	leasePath, lease, err := s.Leases.GetLease(ctx, tokenStr)
 	if err != nil {
 		outcome = err.Error()
-		return err
+		return 0, err
 	}
 
 	if err := CheckToken(tokenStr, lease.Token.Secret); err != nil {
 		outcome = err.Error()
-		return err
+		return 0, err
 	}
 
 	repository, _, err := gw.SplitLeasePath(leasePath)
 	if err != nil {
 		outcome = err.Error()
-		return err
+		return 0, err
 	}
+	var finalRev uint64
 	if err := s.Leases.WithLock(ctx, repository, func() error {
-		return s.Pool.CommitLease(ctx, leasePath, oldRootHash, newRootHash, tag)
+		var err error
+		finalRev, err = s.Pool.CommitLease(ctx, leasePath, oldRootHash, newRootHash, tag)
+		return err
 	}); err != nil {
 		outcome = err.Error()
-		return err
+		return 0, err
+	}
+
+	plotsErr := s.StatsMgr.UploadStatsPlots(repository)
+	if plotsErr != nil {
```

Comment:
I am thinkong about running this is a goroutine, but I believe there could be race conditions, right?

Reply:
That's actually a good idea! It takes a long time and we don't need its result for anything else. The script should be kinda race-less.